### DISCUSSION TITLE
Faster autostrafe turning for better user control

### DISCUSSION THREAD

**Discussion by SeasonalFerret** [Fedoraware/Fedoraware#973] [Ideas]
I love fware's quick-stop and fast-accel, makes me feel in control of my movement. But they do nothing while mid-air. To be quite frank I've never felt so fluid in tf2 movement before using rijin's autostrafe. Is it possible to mimic rijin's behaviour at all?

On rijin, getting knocked into the air by a soldier as a spy means that I essentially fly straight towards the soldier with how fast it turns. On fware, I'm constantly over-shooting jumps to the point where autostrafe is more of a nuisance than anything.

Rijin appears to have something called "retrack speed" that they use to set how fast it autostrafes into the WASD direction you're holding.

rijin's:

https://user-images.githubusercontent.com/17802843/188365781-983ad9b1-db1e-469c-b38f-4ae9776ab34d.mp4

https://user-images.githubusercontent.com/17802843/188365786-37e6ecea-197d-4ef4-8913-61fe6c2f42a0.mp4

fedora's:

https://user-images.githubusercontent.com/17802843/188365771-7ccba948-a430-4247-9702-10fa3c8a8bde.mp4

https://user-images.githubusercontent.com/17802843/188366128-ca002439-5bbf-4de9-82cb-1f1f946ffe6b.mp4

**Comment by HutchyBen** [comment]
Ask Mr rijin.solitions to contribute a pull request

**Comment by canedoly** [comment]
there's a +strafe command in tf2, which allows you to move in the direction your mouse is using which also works in air, its also dependent by how fast your mouse is moving (if this is the same method as rijin, it explains why they have retrack speed) so this could be a method to improve auto strafe, but idk how to implement it and if it works

**Comment by lnd1g0** [comment]
> there's a +strafe command in tf2, which allows you to move in the direction your mouse is using which also works in air, its also dependent by how fast your mouse is moving (if this is the same method as rijin, it explains why they have retrack speed) so this could be a method to improve auto strafe, but idk how to implement it and if it works

This was a leftover movement option from Quake, which was carried over from Doom. In Doom, you have the option to move around with the mouse instead of the keyboard, and you could use the arrow keys to look left and right.

**Comment by SeasonalFerret** [comment]
Seems like we could sacrifice speed to increase the turning radius

**Comment by johnathon-walnut** [comment]
im too stupid and i dont know how to do this :)

**Comment by SeasonalFerret** [comment]
Maybe keep it open in case someone else figures out how to do it?

I was working on it for a bit.
